README.md
# Sightstone Project
Sightstone is an open-source and alternative League of Legends client. It is currently being rewritten in order to organize code and provide a better user experience. 
It is aimed to all sort of setups, letting you choose from the below four options:
  * Sightstone Plus
  * Sightstone
  * Sightstone Light
  * Sightstone Ultra Light

# Sightstone Plus
Sightstone Plus comes extras feature not present in Sightstone (see below) such as:
  * Allows you to link social media and dropbox for sharing of replays and screenshots
  * Has a server to allow you to play and chat while away from home and everything will be recorded (Windows, Android, Windows Phone)



# Sightstone
Sightstone comes with a few extra features not present in the original client such as:
  * While in ARAM champions select, you are able to see enemy team champions
  * Record your games automatically and store them
  * Change client theme and UI
  * Store various accounts safely

# Sightstone Light
As mentioned by its name, this is a light version of Sightstone. It has nothing but the original League of Legends client's functions.

# Sightstone Ultra Light
From the three options available this is the lightest version. It comes down to the basic functionality of the League of Legends Client by  removing non-essencial features like:
  * News
  * Custom games
  * Featured games



agsxmpp/changelog.txt
version 1.2
- added agsXMPP.protocol.x.muc.user.Destroy (see: http://forum.ag-software.de/forum.php?req=thread&postid=3663&nocount=1)
- fixed a bug in clientsocket which can cause reconnect problems under some conditions
- fixed Nickname bug in agsXMPP.protocol.extensions.bookmarks.Conference
- added  new constructor to XmppClientConnenction
- added implicit operators to Jid class
- fixed a race condition in Bosh
- added NotModified error condition
- added Errortext property to Error class
- set type automatically to error when an error child gets added
- fixed sasl DIGESTMD5 error with escaped jids
- added some more generic functions to Element
- AutoResolveConnectServer is true by default now
- added GSSAPI SASL mechanism (only enterprise version)

version 1.1
============
- added TLS for Compact Framework
- bugfix in clientsocket.cs which caused reconnect problems in some cases
- some minor bugfixes

version 1.0
============
- added XEP-0153: vCard-Based Avatars
- added bookmark storage XEP-0048
- added message receipts XEP-0184
- added XEP-0047: In-Band Bytestreams (IBB)
- added a hack to not send xmlns="jabbe:client" with all stanzas
- added AutoPresence Property. This sends the own presence in XmppClientConnection automatically after login. If you already
  send your presence after succesful login set AutoPresence to false or remove the SendMyPresence call
- fixed a bug in agsXMPP.protocol.sasl.Mechamism  
- added SASL failure conditions (FailureConditions.cs)
- fixed HasChildElements property in Element.cs
- did some refactoring on Element.cs
- added some members to Element
- added Console Client example
- fixed typos in DiscoManager.cs
- updated caps to teh latest xep version, can create the new ver attrib automatically.
- published first BOSH version. Interoperability only tested with ejabberd2 beta 1
- optimized Polling code
- updated caps to the latest version of the XEP where caps versioning was changed.
- added AMP
- refactored and extended base classes

version 0.95
============
- many improvements and bugfixes
- added more stuff to the MucManager

version 0.93
============
- improvements on socket errors
- imporvements in resource binding. Leave resource empty and the server will asign a resource for you.
- Added ClientName and ClientFeatures
- Automatic response to DiscoInfo requests to clients

version 0.93
============
- improvements in XmppComponentConnection
- improved Keep Alive logic
- OnXmppErrorHandler -> XmppElementHandler
- XmppElementHandler -> XmppStanzaHandler ???
- refactored IqHandler event public event (StreamHandler ==> IqHandler)
- CreateNewThread member added for Messages
- EscapeNode and Unescape Node in the Jid class is now static
- added stringprep support and libidn#
- fixes in jivesoftware phone stuff

version 0.92
============
- added XEP-0115 Entity Capabilities
- added XEP-0106 JID Escaping
- added XEP-0199 XMPP Ping
- licence change

version 0.91
============
- fixed register account for servers which support only sasl Login
- fixed a bug in ClientSocket.cs with the connect timeout timer
- added X-GOOGLE-TOKEN authentication and Sasl mechanism
  This means you can connect also on PPC and Smartphone to GTalk, or without SSL/TLS with Desktop Edition to GTalk
- added synchronous Packet Grabbing to the IqGrabber
- added new enum for packetId types. Now you can use Guid's for packet ID's

version 0.90
============
- improved srv record handling
- added nickname support (JEP-0172 User nickname)
- imporoved filetransfer
- fixed SASL to work with jabberd 1.4x
- added disco manager
- lots of minor changes and improvements

version 0.81
============
- minor changes and refactoring
- added bytestream support

version 0.80
============
- added Ad-Hoc Commands
- added pubsub
- some minor changes and refactoring

version 0.77
============
- added privacy list support (XMPP Core)

version 0.76
============
- fixed a bug in XmppPacket (Error property)
- fixed a bug in Register.cs
- added Geoloc
- started to add Jingle
- small bugfix in event.cs
- optimized the vcard code

version 0.75
============
- added MUC
- Chatstates JEP-0085
- Added RPC JEP-0009 (Transporting XML-RPC over Jabber)
- Added ElementList and changed all Select* functions that returned NodeList to the return value ElementList, because the NodeList messed up indexes

version 0.71
============
- added SASL ANONYMOUS
- fixed a bug in Mechanism.cs (password was always lower case)

version 0.70
============
- 1st version of new roster control in agsXMPP.ui
- stream compression
- many improvements an minor bugfixes
- some namespaces changed a bit because of VB .NET programmers
- new version of Mono.Security.dll for .NET 1.1 included (see: http://bugzilla.ximian.com/show_bug.cgi?id=75687)

version 0.65
============
- renamed featues namespace to feature becuase of VB programmers that couldn't access it
- added stream compression and stream compression classes

version 0.63
============
- added <item/> and <reported/> to xdata (for xdata search)
- fixed a entity bug in attributes (StreamParser.cs) 

version 0.62
============
- added PresenceGrabber to XmppClientConnection
- added a new overload to the Helper functions in PresenceManager

version 0.61
============
- Bugfix in Disco feature and identity

version 0.60
============
- switched to visual studio 2005 and .NET 2.0
- added SSL support for the build in SSL Stream in .NET 2.0
- added support for Jive Software Phone Extensions

version 0.51
============
- added BrowseItem class
- changes in Browse to support historical service protocol
- fixes in OnRosterStart and OnRosterEnd event

version 0.50
============
- changed parser to xpnet
- added xdata classes and xdata controls
- added PresenceManager
- added RosterManager
- added private xml storage (JEP-0049)
- added nested roster groups (JEP-0083)
- rewrote StreamParser.cs
- rewrote ClientSocket.cs

version 0.40
============
- added Start-TLS Support
- added Service Discovery (JEP-0030)

agsxmpp/Uri.cs
/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
 * Copyright (c) 2003-2016 by AG-Software 											 *
 * All Rights Reserved.																 *
 * Contact information for AG-Software is available at http://www.ag-software.de	 *
 *																					 *
 * Licence:																			 *
 * The agsXMPP SDK is released under a dual licence									 *
 * agsXMPP can be used under either of two licences									 *
 * 																					 *
 * A commercial licence which is probably the most appropriate for commercial 		 *
 * corporate use and closed source projects. 										 *
 *																					 *
 * The GNU Public License (GPL) is probably most appropriate for inclusion in		 *
 * other open source projects.														 *
 *																					 *
 * See README.html for details.														 *
 *																					 *
 * For general enquiries visit our website at:										 *
 * http://www.ag-software.de														 *
 * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */

using System;

namespace agsXMPP
{
	
	public class Uri
	{
		public const string STREAM			= "http://etherx.jabber.org/streams";
		public const string CLIENT			= "jabber:client";		
		public const string SERVER			= "jabber:server"; 
		
		public const string IQ_AGENTS		= "jabber:iq:agents";
		public const string IQ_ROSTER		= "jabber:iq:roster";
		public const string IQ_AUTH			= "jabber:iq:auth";
		public const string IQ_REGISTER		= "jabber:iq:register";
		public const string IQ_OOB			= "jabber:iq:oob";
		public const string IQ_LAST			= "jabber:iq:last";
		public const string IQ_TIME			= "jabber:iq:time";
		public const string IQ_VERSION		= "jabber:iq:version";
		public const string IQ_BROWSE		= "jabber:iq:browse";
		public const string IQ_SEARCH		= "jabber:iq:search";
		public const string IQ_AVATAR		= "jabber:iq:avatar";
		public const string IQ_PRIVATE		= "jabber:iq:private";
        public const string IQ_PRIVACY      = "jabber:iq:privacy";


        
        /// <summary>
        /// JEP-0009: Jabber-RPC
        /// </summary>
        public const string IQ_RPC          = "jabber:iq:rpc";
        
		
		public const string X_DELAY			= "jabber:x:delay";
		public const string X_EVENT			= "jabber:x:event";
		public const string X_AVATAR		= "jabber:x:avatar";
		
        
		public const string X_CONFERENCE	= "jabber:x:conference";
		
        /// <summary>
        /// jabber:x:data
		/// </summary>
        public const string X_DATA			= "jabber:x:data";
		
		/// <summary>
		/// JEP-0144 Roster Item Exchange
		/// </summary>
		public const string X_ROSTERX		= "http://jabber.org/protocol/rosterx";
		

		/// <summary>
        /// Multi User Chat (MUC) JEP-0045
        /// http://jabber.org/protocol/muc
		/// </summary>
		public const string MUC				= "http://jabber.org/protocol/muc";
		/// <summary>
		/// http://jabber.org/protocol/muc#user
		/// </summary>
		public const string MUC_USER		= "http://jabber.org/protocol/muc#user";
		/// <summary>
		/// "http://jabber.org/protocol/muc#admin
		/// </summary>
		public const string MUC_ADMIN		= "http://jabber.org/protocol/muc#admin";
		/// <summary>
		/// http://jabber.org/protocol/muc#owner
		/// </summary>
		public const string MUC_OWNER		= "http://jabber.org/protocol/muc#owner";

		// Service Disovery
		public const string DISCO_ITEMS		= "http://jabber.org/protocol/disco#items";		
		public const string DISCO_INFO		= "http://jabber.org/protocol/disco#info";

		public const string STORAGE_AVATAR	= "storage:client:avatar";

		public const string VCARD			= "vcard-temp";

		// New XMPP Stuff
        /// <summary>
        /// urn:ietf:params:xml:ns:xmpp-streams
        /// </summary>
        public const string STREAMS         = "urn:ietf:params:xml:ns:xmpp-streams";
        public const string STANZAS			= "urn:ietf:params:xml:ns:xmpp-stanzas";
		public const string TLS				= "urn:ietf:params:xml:ns:xmpp-tls";		
		public const string SASL			= "urn:ietf:params:xml:ns:xmpp-sasl";		
		public const string SESSION			= "urn:ietf:params:xml:ns:xmpp-session";		
		public const string BIND			= "urn:ietf:params:xml:ns:xmpp-bind";
        

        /// <summary>
        /// jabber:component:accept
        /// </summary>
		public const string ACCEPT			= "jabber:component:accept";

		// Features
		//<register xmlns='http://jabber.org/features/iq-register'/>
		public const string FEATURE_IQ_REGISTER	= "http://jabber.org/features/iq-register";
        /// <summary>
        /// Stream Compression http://jabber.org/features/compress
        /// </summary>
        public const string FEATURE_COMPRESS    = "http://jabber.org/features/compress";

		// Extensions (JEPs)
		public const string SHIM				= "http://jabber.org/protocol/shim";
		public const string PRIMARY				= "http://jabber.org/protocol/primary";
        /// <summary>
        /// JEP-0172 User nickname
        /// http://jabber.org/protocol/nick
        /// </summary>
        public const string NICK                = "http://jabber.org/protocol/nick";
                
        /// <summary>
        /// JEP-0085 Chat State Notifications
        /// http://jabber.org/protocol/chatstates
        /// </summary>
        public const string CHATSTATES          = "http://jabber.org/protocol/chatstates";

        /// <summary>
        /// JEP-0138: Stream Compression
        /// </summary>
        public const string COMPRESS            = "http://jabber.org/protocol/compress";
		
		/// <summary>
		/// JEP-0020: Feature Negotiation http://jabber.org/protocol/feature-neg
		/// </summary>
		public const string FEATURE_NEG			= "http://jabber.org/protocol/feature-neg";

		/// <summary>
		/// JEO-0095 http://jabber.org/protocol/si
		/// </summary>
		public const string SI					= "http://jabber.org/protocol/si";
		/// <summary>
		/// JEO-0096 http://jabber.org/protocol/si/profile/file-transfer
		/// </summary>
		public const string SI_FILE_TRANSFER	= "http://jabber.org/protocol/si/profile/file-transfer";

        /// <summary>
        /// JEP-0065 SOCKS5 bytestreams
        /// http://jabber.org/protocol/bytestreams
        /// </summary>
        public const string BYTESTREAMS         = "http://jabber.org/protocol/bytestreams";

		// JEP-0083
		public const string ROSTER_DELIMITER	= "roster:delimiter";

        // Jive Software Namespaces

        /// <summary>
        /// Jivesoftware asterisk-im extension (http://jivesoftware.com/xmlns/phone);
        /// </summary>
        public const string JIVESOFTWARE_PHONE  = "http://jivesoftware.com/xmlns/phone";

        /// <summary>
        /// JEP-0071: XHTML-IM (http://jivesoftware.com/xmlns/phone)
        /// </summary>
        public const string XHTML_IM            = "http://jabber.org/protocol/xhtml-im";
        public const string XHTML			    = "http://www.w3.org/1999/xhtml";

        
        /// <summary>
        /// XEP-0115: Entity Capabilities (http://jabber.org/protocol/caps)
        /// </summary>
        public const string CAPS                = "http://jabber.org/protocol/caps";

        /// <summary>
        /// Jingle http://jabber.org/protocol/jingle
        /// </summary>
        public const string JINGLE                  = "http://jabber.org/protocol/jingle";

        /// <summary>
        /// Jingle audio format description http://jabber.org/protocol/jingle/description/audio
        /// </summary>
        public const string JINGLE_AUDIO_DESCRIPTION = "http://jabber.org/protocol/jingle/description/audio";

        /// <summary>
        /// Jingle Info audio http://jabber.org/protocol/jingle/info/audio;
        /// </summary>
        public const string JINGLE_AUDIO_INFO        = "http://jabber.org/protocol/jingle/info/audio";


        public const string JINGLE_VIDEO_DESCRIPTION = "http://jabber.org/protocol/jingle/description/video";

        /// <summary>
        /// GeoLoc (http://jabber.org/protocol/geoloc)
        /// </summary>
        public const string GEOLOC              = "http://jabber.org/protocol/geoloc";

        /// <summary>
        /// <para>XMPP ping</para>
        /// <para>Namespace: urn:xmpp:ping</para>
        /// <para><seealso cref="http://www.xmpp.org/extensions/xep-0199.html">http://www.xmpp.org/extensions/xep-0199.html</seealso></para>
        /// </summary>
        public const string PING                = "urn:xmpp:ping";
                
        /// <summary>
        /// Ad-Hoc Commands (http://jabber.org/protocol/commands)
        /// </summary>
        public const string COMMANDS            = "http://jabber.org/protocol/commands";

        // Pubsub stuff
        public const string PUBSUB              = "http://jabber.org/protocol/pubsub";
        public const string PUBSUB_EVENT        = "http://jabber.org/protocol/pubsub#event";        
        public const string PUBSUB_OWNER        = "http://jabber.org/protocol/pubsub#owner";

        // Http-Binding XEP-0124
        public const string HTTP_BIND           = "http://jabber.org/protocol/httpbind";
        
        /// <summary>
        /// <para>XEP-0184: Message Receipts</para>
        /// <para>urn:xmpp:receipts</para>        
        /// </summary>
        public const string MSG_RECEIPT         = "urn:xmpp:receipts";

        /// <summary>
        /// <para>XEP-0048: Bookmark Storage</para>
        /// <para>storage:bookmarks</para>
        /// </summary>
        public const string STORAGE_BOOKMARKS   = "storage:bookmarks";

        /// <summary>
        /// <para>XEP-0047: In-Band Bytestreams (IBB)</para>
        /// <para>http://jabber.org/protocol/ibb</para>
        /// </summary>
        public const string IBB                 = "http://jabber.org/protocol/ibb";

        /// <summary>
        /// <para></para>
        /// <para>http://jabber.org/protocol/amp</para>
        /// </summary>
        public const string AMP                 = "http://jabber.org/protocol/amp";

        /// <summary>
        /// <para>XEP-0153: vCard-Based Avatars</para>
        /// <para>vcard-temp:x:update</para>
        /// </summary>
        public const string VCARD_UPDATE        = "vcard-temp:x:update";

        public const string URN_TIME            = "urn:xmpp:time";

        /// <summary>
        /// <para>XEP-0145 Annotations</para>
        /// <para>storage:rosternotes</para>
        /// </summary>
        public const string STORAGE_ROSTERNOTES = "storage:rosternotes";
	}	
}